addons/mail/wizard/mail_compose_message.py

```diff
@@ -63,17 +63,25 @@ def default_get(self, fields):
             if missing_author:
                 result['author_id'] = author_id
 
-        result['composition_mode'] = result.get('composition_mode', 'comment')
-        result['model'] = result.get('model', self._context.get('active_model'))
-        result['res_id'] = result.get('res_id', self._context.get('active_id'))
-        if 'no_auto_thread' not in result and (result['model'] not in self.env or not hasattr(self.env[result['model']], 'message_post')):
+        if 'composition_mode' in fields and 'composition_mode' not in result:
+            result['composition_mode'] = result.get('composition_mode', 'comment')
+        if 'model' in fields and 'model' not in result:
+            result['model'] = result.get('model', self._context.get('active_model'))
+        if 'res_id' in fields and 'res_id' not in result:
+            result['res_id'] = result.get('res_id', self._context.get('active_id'))
+        if 'no_auto_thread' in fields and 'no_auto_thread' not in result and 'model' in fields and (result.get('model') not in self.env or not hasattr(self.env[result['model']], 'message_post')):
             result['no_auto_thread'] = True
 
         vals = {}
         if 'active_domain' in self._context:  # not context.get() because we want to keep global [] domains
             vals['active_domain'] = '%s' % self._context.get('active_domain')
-        if result['composition_mode'] == 'comment':
-            vals.update(self.get_record_data(result))
+        if result.get('composition_mode') == 'comment' and any(fname in fields for fname in ['model', 'res_id', 'partner_ids', 'record_name', 'subject']):
```

Comment:
Could use set intersection:
`and set(fields) & set(['model', 'res_id', 'partner_ids', 'record_name', 'subject'])`

But maybe it's a bit less readable, I let you choose.